src/Analyzer/CodeAnalyzer.cs

```diff
@@ -149,8 +149,17 @@ public ImmutableArray<DiagnosticAnalyzer> GetAnalyzers()
                 if (File.Exists(sonarConfigurationPath))
                 {
                     var xmlDoc = XDocument.Load(sonarConfigurationPath);
-                    PatternIds = xmlDoc.Descendants("Rule").Select(e => e.Elements("Key").Single().Value)
-                        .ToImmutableList();
+                    var analysisInput = xmlDoc.Element("AnalysisInput");
+                    var rules = analysisInput.Element("Rules"); 
+                    if (rules is null)
+                    {
+                        return availableAnalyzers;
+                    }
+                    else
+                    {
+                        PatternIds = xmlDoc.Descendants("Rule").Select(e => e.Elements("Key").Single().Value)
```

Comment:
Maybe you could use `rules` instead of `xmlDoc` and replace `Descendants` method with `Elements`. This way, instead of searching on the entire xml doc tree, you only iterate over the rules.

Reply:
Absolutely true! I forgot to do it.